### PR TITLE
mummer: add subcommands and integrate into single file

### PR DESCRIPTION
**Pull request by cgjosephlee**
Add subcommands and integrate into single file, also add a version detector.

### CODE REVIEW

File: completions/_mummer [hunk 1]
```diff
@@ -1,26 +1,315 @@
...
+        '(-maxmatch -mum -mumreference -mumcand)'{-mumreference,-mumcand}'[compute maximal matches that are unique in the reference but not necessarily in the query-sequence (default)]'
+        '(-n)-n[match only the characters a, c, g, or t; they can be in upper or in lower case]'
+        '(-b -r)-r[only compute reverse complement matches]'
+        '(-s)-s[show the matching substrings]'
+        '(-v --version)'{-v,--version}'[Display the version information and exit]'
+        '*:file:_files'
+    )
+    if [ "$nuc_ver" -eq 4 ] ; then
+        args+=(
+            '-k[sampled suffix positions (1)]'
```
Comment (kloetzl) [inline completions/_mummer]: The `(1)` make me think that `-k` expects an argument.

**Comment by kloetzl** [comment]
Thank you for your work. I noticed a few typos: s/folat/float/, s/idenitity/identity/. Also, please be consistent with case: Either begin every description with a lower case or upper case character.

**Comment by cgjosephlee** [comment]
Sorry for carelessness, hope this commit goes fine.

**Comment by kloetzl** [comment]
Here is one other thing I realised: In exclusion lists you tend to write the long option with a `=` if it takes a parameter. Eg. `'(-t --threads=)'{-t,--threads=-}'[Use NUM threads (default=# of cores)]:int:'`. As far as I can tell, that doesn't work. Make it just `'(-t --threads)'`.

**Comment by cgjosephlee** [comment]
Thank you. The equal signs in parentheses are now removed.

**Comment by kloetzl** [comment]
Thank you very much, merged. :fireworks:

**Pull request merged**